packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts

```diff
@@ -236,6 +236,18 @@ export const useGridFocus = (
     [apiRef],
   );
 
+  const handleCleanFocus = React.useCallback<GridEventListener<'rowsSet'>>(() => {
+    const cell = gridFocusCellSelector(apiRef);
+
+    // If the focused cell is in a row which does not exist anymore, then remove the focus
+    if (cell && !apiRef.current.getRow(cell.id)) {
+      apiRef.current.setState((state) => ({
+        ...state,
+        focus: { cell: null, columnHeader: null },
```

Comment:
In theory, every time a cell loses focus it should publish `cellFocusOut`, but to publish it we call `getCellParams` which will crash because the row doesn't exist. It will be a bug if the removed row was also being edited. If we want to implement the option 3 from https://github.com/mui/mui-x/issues/4516#issuecomment-1102916209 we will likely have to revisit this part. Maybe we change the params passed to `cellFocusOut` to not include the row object. Anyway, until someone complains about `cellFocusOut` not published we can keep the listener to `rowsSet`.

Reply:
Yes, in v6 we could pass the id instead of the params to allow to publish the focus out event on a destroyed cell.